src/app/core/product-form.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import {priceRegEx} from '../utils/validation';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductFormService {
+
+  constructor(private fb: FormBuilder) {}
+
+  generateFormValidation(name = '', photoUrl = '', price = 0, productDescription = '', flagged = false) {
```

Comment:
Consider using a single parameter for this method. `generateFormValidation(product?: IProduct)`